### ISSUE TITLE
[java] UnusedPrivateMethod violation for disabled class in 6.23.0

### ISSUE THREAD

**Issue by mmoehring** [pmd/pmd#2454]
**Affects PMD Version:** 6.23.0

**Rule:**
UnusedPrivateMethod

https://pmd.github.io/pmd-6.23.0/pmd_rules_java_bestpractices.html#unusedprivatemethod

**Description:**
In 6.22.0 the rule uses the classes listed in the `ignoredAnnotations`property as expected. In 6.23.0 I'm getting a violation for the class `javax.annotation.PreDestroy`on a private method.

**Code Sample demonstrating the issue:**

```Java
import javax.annotation.PreDestroy;

@PreDestroy
private void shutdown() {
    server.shutdown();
}
```

**Ruleset:**
```XML
<rule ref="category/java/bestpractices.xml/UnusedPrivateMethod">
    <properties>
        <property name="ignoredAnnotations" value="java.lang.Deprecated|javax.annotation.PostConstruct|
            javax.annotation.PreDestroy|org.springframework.context.event.EventListener|
            org.springframework.scheduling.annotation.Scheduled"/>
    </properties>
</rule>
```

**Expected outcome:**
The private method annotated with `@PreDestroy` should not be a violation.

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *Gradle*

**Comment by KroArtem** [comment]
I tried to dig into this bug but failed halfway.

If you define only your `javax.annotation.PreDestroy` annotation in `ignoredAnnotations` or if you define them all in one line, everything would be ok.

I've reduced the number of annotations and here is what I see in debug
<img width="744" alt="Screen Shot 2020-05-13 at 19 24 07" src="https://user-images.githubusercontent.com/1813101/81838895-75d3ef00-954f-11ea-95fc-8606d270b328.png">

That's why your annotations is not found in `propertyValuesByDescriptor`.

This is where I lost the track. Maybe with a guidance from @adangel or @oowekyala we'll find the cause.

**Comment by adangel** [comment]
That maybe means, we would need to trim the property values. Currently we get the attribute value and split by the delimiter (in that case `|`),  but we don't trim.
I'm not sure, whether we can trim in all cases - what if the whitespace is important? (in that case, for a list of class names, surely not, but for other properties maybe....)

**Comment by KroArtem** [comment]
@adangel , if you'd point me where properties are read and parsed, I think I'll be able to continue digging into it. As @mmoehring says, it was working in 6.22 so it's probably a regression.

**Comment by KroArtem** [comment]
@adangel , I've tried to look through 6.23-6.22 changes but didn't manage to find any suspicious code. If you can give a hint where properties are read, I can continue with this issue.

**Comment by adangel** [comment]
I also couldn't find any changes related to properties. So, first step: reproduce it... can you reproduce it with PMD 6.22.0? So, double check, whether that's really a regression...

**Comment by oowekyala** [comment]
Looks like this was caused by #2380, and probably more specifically this:

https://github.com/pmd/pmd/pull/2380/files#diff-2721d59031040d9cd3b2e2b43dc5a76972de458ffd96db20821a94bdd2a36ce0R84

Before that pr there was a single path to load the class, that used ClassUtils.getClass:

https://github.com/pmd/pmd/blob/17ddfd2a22eb815dfbb4346ec838ff8117a0a224/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/TypeHelper.java#L119

which starts by deleting whitespace in the class name.
